R/methods-Links.R

```diff
@@ -0,0 +1,32 @@
+# Methods for Link and LinkList objects
+
+# Accessors for fanciness
+setGeneric("source", function(object) standardGeneric("source"))
+setGeneric("source<-", function(object, value) standardGeneric("source<-"))
+setGeneric("target", function(object) standardGeneric("target"))
+setGeneric("target<-", function(object, value) standardGeneric("target<-"))
+setGeneric("weight", function(object) standardGeneric("weight"))
+setGeneric("weight<-", function(object, value) standardGeneric("weight<-"))
+setGeneric("color", function(object) standardGeneric("color"))
+setGeneric("color<-", function(object, value) standardGeneric("color<-"))
+
+setMethod("source", "Link", function(object) object@source)
+setMethod("source<-", "Link", function(object, value) {object@source <- value; validObject(object); object})
+setMethod("target", "Link", function(object) object@target)
+setMethod("target<-", "Link", function(object, value) {object@target <- value; validObject(object); object})
+setMethod("weight", "Link", function(object) object@weight)
+setMethod("weight<-", "Link", function(object, value) {object@weight <- value; validObject(object); object})
+setMethod("color", "Link", function(object) object@color)
+setMethod("color<-", "Link", function(object, value) {object@color <- value; validObject(object); object})
+
+
+# For LinkLists, let's return vectors of data from the nodes
+setGeneric("getSourceNodes", function(object) standardGeneric("getSourceNodes"))
+setMethod("getSourceNodes", "LinkList", function(object) lapply(object, function(x) source(x)))
```

Comment:
Could be a question here whether this would be expected to be a list of nodes vs a nodeList

Reply:
updated. I'm leaving as a list of nodes for now because i can't imagine we always really want a NodeList. But maybe in the next few weeks of using it more i'll change my mind.